File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer, Integer> prefixSum = new HashMap<>();
        prefixSum.put(0, 1);
        
        int totalSum = 0, subArrayCount = 0, difference = 0;

        for (int num : nums) {
          totalSum += num;
          difference = totalSum - k;
          subArrayCount += prefixSum.getOrDefault(difference, 0);
          prefixSum.put(totalSum, prefixSum.getOrDefault(totalSum, 0) + 1);
        }

        return subArrayCount;
    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        perfix_sum = {0: 1}
        total_sum = sub_array_count = 0

        for num in nums:
          total_sum += num
          difference = total_sum - k
          sub_array_count += perfix_sum.get(difference, 0)
          perfix_sum[total_sum] = perfix_sum.get(total_sum, 0) + 1

        return sub_array_count